### PR TITLE
bikeshed: AUTO_ALLOW_WAYPOINT_POLICY

### PR DESCRIPTION
**Pull request by stevenctl**
The name as it stands is a bit confusing.

### CODE REVIEW

**Review by howardjohn — APPROVED** [review]

**Review by linsun — COMMENTED** [review]
LGTM

cc @costinm

File: pilot/pkg/features/ambient.go [hunk 1]
```diff
@@ -46,9 +46,10 @@ var (
 		"If enabled, HBONE will be preferred when sending to destinations. ")
 
 	DefaultAllowFromWaypoint = registerAmbient(
-		"PILOT_DEFAULT_ENFORCE_FROM_WAYPOINT",
+		"PILOT_AUTO_ALLOW_WAYPOINT_POLICY",
 		true, false,
-		"If enabled, zTunnels will enforce that incoming traffic has traversed a Waypoint if traffic to an endpoint may traverse one.")
+		"If enabled, zTunnel's will receive synthetic authorization policies for each workload ALLOW the Waypoint's identity. "+
```
Comment (stevenctl) [inline pilot/pkg/features/ambient.go:51]:
```suggestion
		"If enabled, zTunnel will receive synthetic authorization policies for each workload ALLOW the Waypoint's identity. "+
```

File: pilot/pkg/features/ambient.go [hunk 2]
```diff
@@ -46,9 +46,10 @@ var (
 		"If enabled, HBONE will be preferred when sending to destinations. ")
 
 	DefaultAllowFromWaypoint = registerAmbient(
-		"PILOT_DEFAULT_ENFORCE_FROM_WAYPOINT",
+		"PILOT_AUTO_ALLOW_WAYPOINT_POLICY",
```
Comment (keithmattix) [inline pilot/pkg/features/ambient.go:49]: IIUC, the polarity of the flag changed but the default didn't?

Comment (linsun) [reply pilot/pkg/features/ambient.go:49]: Correct.  We do need to make a decision to change to enable the flag before 1.22 goes out.

Comment (keithmattix) [reply pilot/pkg/features/ambient.go:49]: @stevenctl Was this decision made?

Comment (linsun) [reply pilot/pkg/features/ambient.go:49]: I don't think so. I personally would like to test a bit more as well.

**Pull request merged**